src/components/calcite-input-time-picker/calcite-input-time-picker.tsx

```diff
@@ -0,0 +1,330 @@
+    if (this.parseTimeString(event.detail.value) !== null || !event.detail.value) {
+      this.value = event.detail.value;
+    }
+  };
+
+  @Listen("keydown")
+  inputKeyDownHandler(event: KeyboardEvent): void {
+    // This prevents the browser default time picker UI from appearing
+    if (
+      (event.target as HTMLElement).closest("calcite-input") === this.inputEl &&
+      event.key === " "
+    ) {
+      event.preventDefault();
+    }
+  }
+
+  @Listen("keyup")
+  keyUpHandler(event: KeyboardEvent): void {
+    if (event.key === "Escape" && this.popoverOpen === true) {
+      this.popoverOpen = false;
+    }
+  }
+
+  @Listen("calciteTimePickerBlur")
+  timePickerBlurHandler(event: CustomEvent): void {
+    event.preventDefault();
+    event.stopPropagation();
+    this.popoverOpen = false;
+  }
+
+  @Listen("calciteTimePickerChange")
+  timePickerChangeHandler(event: CustomEvent): void {
+    event.preventDefault();
+    event.stopPropagation();
+    if (event.detail) {
+      const { hour, minute, second } = event.detail as Time;
+      if (hour !== "--" && minute !== "--") {
+        if (this.step !== 60 && second !== "--") {
+          this.value = `${hour}:${minute}:${second}`;
+        } else {
+          this.value = `${hour}:${minute}`;
+        }
+      } else {
+        this.value = "";
+      }
+    }
+  }
+
+  @Listen("calciteTimePickerFocus")
+  timePickerFocusHandler(event: CustomEvent): void {
+    event.preventDefault();
+    event.stopPropagation();
+    this.popoverOpen = true;
+  }
+
+  @Listen("click", { target: "window" })
+  windowClickHandler(event: MouseEvent): void {
+    const target = event.target as HTMLElement;
+    const closestHost = target.closest(
+      "calcite-input-time-picker"
+    ) as HTMLCalciteInputTimePickerElement;
+    const closestLabel = target.closest("calcite-label") as HTMLCalciteLabelElement;
+    if (closestLabel && closestLabel.for === this.guid) {
+      this.inputEl.setFocus();
+      this.popoverOpen = true;
+    } else if (closestHost !== this.el) {
+      this.popoverOpen = false;
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method()
+  async setFocus(): Promise<void> {
+    this.inputEl.setFocus();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private convertStringToTime = (value: string): Time => {
+    const timeString = this.parseTimeString(value);
+    const [hour, minute, second] = timeString ? timeString.split(":") : ["--", "--", "--"];
+    return {
+      hour,
+      minute,
+      second: second || (hour !== "--" && minute !== "--" ? "00" : "--")
+    };
+  };
+
+  private stringContainsOnlyNumbers(string): boolean {
```

Comment:
Maybe some of these functions that are just handling parsing/logic should be moved into a utils file?

Reply:
Definitely, I'm seeing the need to use this function inside time-picker anyway.  Will update that real quick.

Reply:
Sounds like it might also be a good idea to move my `time-picker/utils.ts` into `src/utils/time.ts` to match what date-picker is doing.  I'll do that as well.